cirq-google/cirq_google/engine/engine_client_test.py

```diff
@@ -851,6 +847,59 @@ def test_run_job_over_stream(
     stream_manager.submit.assert_called_with(*expected_submit_args)
 
 
+@pytest.mark.parametrize(
+    'run_job_kwargs, expected_submit_args',
+    [
+        (
+            {
+                'project_id': 'proj',
+                'program_id': 'prog',
+                'code': any_pb2.Any(),
+                'job_id': 'job0',
+                'processor_id': 'processor0',
+                'run_context': any_pb2.Any(),
+                'snapshot_id': 'SNAPSHOT_ID',
+                'device_config_name': 'CONFIG_NAME',
+            },
+            [
+                'projects/proj',
+                quantum.QuantumProgram(name='projects/proj/programs/prog', code=any_pb2.Any()),
+                quantum.QuantumJob(
+                    name='projects/proj/programs/prog/jobs/job0',
+                    run_context=any_pb2.Any(),
+                    scheduling_config=quantum.SchedulingConfig(
+                        processor_selector=quantum.SchedulingConfig.ProcessorSelector(
+                            processor='projects/proj/processors/processor0',
+                            device_config_selector=quantum.DeviceConfigSelector(
+                                snapshot_id="SNAPSHOT_ID", config_alias="CONFIG_NAME"
+                            ),
+                        )
+                    ),
+                ),
+            ],
+        )
+    ],
+)
+@mock.patch.object(quantum, 'QuantumEngineServiceAsyncClient', autospec=True)
+@mock.patch.object(engine_stream_manager, 'StreamManager', autospec=True)
+def test_run_job_over_stream_with_snapshot_id_passes(
+    manager_constructor, client_constructor, run_job_kwargs, expected_submit_args
+):
+    _setup_client_mock(client_constructor)
+    stream_manager = _setup_stream_manager_mock(manager_constructor)
+
+    result = quantum.QuantumResult(parent='projects/proj/programs/prog/jobs/job0')
+    expected_future = duet.AwaitableFuture()
```

Comment:
Since this future is immediately resolved and `result` is never re-used, it could be simplified and remove mutation using [`completed_future`](https://github.com/google/duet/blob/6817af592ecdf2aac0565a0de62bd7287fc9203a/duet/futuretools.py#L203), e.g.:

```py
expected_future = ft.completed_future(quantum.QuantumResult(parent='projects/proj/programs/prog/jobs/job0'))
```